src/openzaak/components/zaken/api/viewsets.py

```diff
@@ -281,6 +285,32 @@ def _zoek(self, request, *args, **kwargs):
 
     _zoek.is_search_action = True
 
+    def get_serializer_context(self):
+        context = super().get_serializer_context()
+        return {
+            **context,
+            "generated_identificatie": self._generated_identificatie,
+        }
+
+    def create(self, request, *args, **kwargs):
+        # Override the parent create to generate an identification _if needed_. The
+        # super create call runs in its own transaction (from NotificationViewSetMixin)
+        # and if needed, we run the identification generation in a separate DB
+        # transaction to mitigate race conditions.
+
+        if not request.data.get("identificatie") and request.data.get(
+            "bronorganisatie"
+        ):
```

Comment:
this is done to favour `ZaakSerializer` for full zaak validation rather than erroring out on just the `bronorganisatie` field, the client correcting that, and then possibly receiving other validation errors for fields not included in the `GenerateZaakIdentificatieSerializer`. So, a better experience for the API client.